lib/std/mem.zig

```diff
@@ -279,15 +279,62 @@ pub fn set(comptime T: type, dest: []T, value: T) void {
 /// Zero initializes the type.
 /// This can be used to zero initialize a C-struct.
```

Comment:
doc comments need updating

Reply:
Also I have a request to add this text:

> Generally, Zig users are encouraged to explicitly initialize all fields of a struct explicitly rather than using this function. However, it is recognized that there are sometimes use cases for initializing all fields to a "zero" value. For example, when interfacing with a C API where this practice is more common and relied upon. If you are performing code review and see this function used, examine closely - it may be a code smell.

Reply:
Will update and add your text.